### PR TITLE
chore(ci): move release jobs to main-cron pipeline

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We want to run main-cron pipeline when we push a new tag and release.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

### CODE REVIEW

**Review by MrCroxx — COMMENTED** [review]
Will main-cron further trigger performance test pipeline?

**Review by lmatz — APPROVED** [review]
lgtm

**Comment by huangjw806** [comment]
> Will main-cron further trigger performance test pipeline?

performance test pipeline run schedule, not triggered by main-cron.

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -581,3 +581,66 @@ steps:
...
+      - "build-amd64"
+      - "build-aarch64"
+    plugins:
+      - seek-oss/aws-sm#v2.3.1:
+          env:
+            GHCR_USERNAME: ghcr-username
+            GHCR_TOKEN: ghcr-token
+            DOCKER_TOKEN: docker-token
+    timeout_in_minutes: 10
+    retry: *auto-retry
```
Comment (kwannoel) [inline ci/workflows/main-cron.yml:646]: Should we run all of these daily?

Comment (huangjw806) [reply ci/workflows/main-cron.yml:646]: This jobs will trigger only build has tag.
```
if: build.tag != null
```

**Review by kwannoel — APPROVED** [review]

**Pull request merged**